### PR TITLE
Container builder

### PR DESCRIPTION
**Pull request by gaoning777**
avoid istio injection 
find out the default namespace

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1774)

### CODE REVIEW

**Comment by gaoning777** [comment]
resolving the namespace issue: https://github.com/kubeflow/pipelines/issues/1775

File: sdk/python/kfp/compiler/_container_builder.py [hunk 1]
```diff
@@ -31,6 +33,10 @@ def __init__(self, gcs_staging, namespace):
       raise ValueError('Error: {} should be a GCS path.'.format(gcs_staging))
     self._gcs_staging = gcs_staging
     self._namespace = namespace
+    if namespace is None:
+      with open(SERVICEACCOUNT_NAMESPACE, 'r') as f:
```
Comment (hongye-sun) [inline sdk/python/kfp/compiler/_container_builder.py]: Could you add a todo to handle not in cluster use case and assume that this code path will only be executed in cluster?

File: sdk/python/kfp/compiler/_container_builder.py [hunk 2]
```diff
@@ -18,11 +18,13 @@
...
+
 class ContainerBuilder(object):
   """
   ContainerBuilder helps build a container image
   """
-  def __init__(self, gcs_staging, namespace):
+  def __init__(self, gcs_staging, namespace=None):
     """
     Args:
       gcs_staging (str): GCS blob that can store temporary build files
```
Comment (hongye-sun) [inline sdk/python/kfp/compiler/_container_builder.py]: add a comment for namespace.

File: sdk/python/kfp/compiler/_container_builder.py (hunk 1 above)
Comment (hongye-sun) [reply sdk/python/kfp/compiler/_container_builder.py]: Or you should handle if the file doesn't exist, and set default value for namespace.

File: sdk/python/kfp/compiler/_container_builder.py (hunk 2 above)
Comment (gaoning777) [reply sdk/python/kfp/compiler/_container_builder.py]: Done.

File: sdk/python/kfp/compiler/_container_builder.py (hunk 1 above)
Comment (gaoning777) [reply sdk/python/kfp/compiler/_container_builder.py]: Good point. Done.

File: sdk/python/kfp/compiler/_container_builder.py [hunk 3]
```diff
@@ -18,19 +18,31 @@
 import os
 import uuid
 
+SERVICEACCOUNT_NAMESPACE = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
```
Comment (Ark-kun) [inline sdk/python/kfp/compiler/_container_builder.py]: Is this universal or does it only work in Kubeflow notebooks?

**Comment by Ark-kun** [comment]
> avoid istio injection

Please create an issue describing the problems we have with Istio. It would be great to prepare for them in other parts of the system.

File: sdk/python/kfp/compiler/_container_builder.py (hunk 3 above)
Comment (gaoning777) [reply sdk/python/kfp/compiler/_container_builder.py]: it happens in k8s clusters. And there is also checks in the code that defaults to 'kubeflow'

**Comment by gaoning777** [comment]
issue created: https://github.com/kubeflow/pipelines/issues/1784

**Pull request merged**